### PR TITLE
feat: add casing option

### PR DESCRIPTION
**Pull request by Shinigami92**
requested in #819

### CODE REVIEW

File: src/modules/random/index.ts [hunk 1]
```diff
@@ -145,21 +151,26 @@ export class Random {
    *
    * @param options Either the number of characters or an options instance. Defaults to `{ count: 1, upcase: false, bannedChars: [] }`.
    * @param options.count The number of characters to generate. Defaults to `1`.
+   * @param options.casing The casing of the characters. Defaults to `'lower'`.
    * @param options.upcase If true, the result will be uppercase. If false, it will be lowercase. Defaults to `false`.
```
Comment (ST-DDT) [inline src/modules/random/index.ts:155]: Maybe remove the default comment and mention that this is deprecated.

File: src/modules/random/index.ts [hunk 2]
```diff
@@ -169,44 +180,40 @@ export class Random {
...
-      's',
-      't',
-      'u',
-      'v',
-      'w',
-      'x',
-      'y',
-      'z',
-    ];
+    let charsArray: string[] = [];
```
Comment (ST-DDT) [inline src/modules/random/index.ts:198]: Instead of pushing it to this array, assign it directly in the case blocks?

Comment (Shinigami92) [reply src/modules/random/index.ts:198]: I want to prevent memory access to the constant readonly arrays. That's why I used spread operators.

Comment (ST-DDT) [reply src/modules/random/index.ts:198]: I meant `charsArray = [...lowerCase]` instead of `charsArray.push()`.

Comment (Shinigami92) [reply src/modules/random/index.ts:198]: Aha! Yeah that could work 🤔

**Comment by Shinigami92** [comment]
ready for review, but maybe I need to add some more tests
lets see what's not covered

**Review by ST-DDT — APPROVED** [review]

File: src/modules/random/index.ts [hunk 3]
```diff
@@ -141,25 +148,30 @@ export class Random {
   }
 
   /**
-   * Generating a string consisting of lower/upper alpha characters based on count and upcase options.
+   * Generating a string consisting of alpha characters.
```
Comment (import-brain) [inline src/modules/random/index.ts:151]:
```suggestion
   * Generating a string consisting of letters in the English alphabet.
```

**Review by import-brain — APPROVED** [review]
Would prefer it to be phrased like this, seems a little bit clearer

Other than that, LGTM

File: src/modules/random/index.ts [hunk 4]
```diff
@@ -141,25 +148,30 @@ export class Random {
...
+   * Generating a string consisting of letters in the English alphabet.
    *
-   * @param options Either the number of characters or an options instance. Defaults to `{ count: 1, upcase: false, bannedChars: [] }`.
+   * @param options Either the number of characters or an options instance. Defaults to `{ count: 1, casing: 'lower', bannedChars: [] }`.
    * @param options.count The number of characters to generate. Defaults to `1`.
-   * @param options.upcase If true, the result will be uppercase. If false, it will be lowercase. Defaults to `false`.
+   * @param options.casing The casing of the characters. Defaults to `'lower'`.
+   * @param options.upcase Deprecated, use `casing: 'upper'` instead.
    * @param options.bannedChars An array with characters to exclude. Defaults to `[]`.
    *
```
Comment (pkuczynski) [inline src/modules/random/index.ts:158]: Idea for the future: `uniq: true`

**Review by pkuczynski — APPROVED** [review]

File: src/modules/random/index.ts [hunk 5]
```diff
@@ -141,25 +148,30 @@ export class Random {
...
 
   /**
-   * Generating a string consisting of lower/upper alpha characters based on count and upcase options.
+   * Generating a string consisting of letters in the English alphabet.
    *
-   * @param options Either the number of characters or an options instance. Defaults to `{ count: 1, upcase: false, bannedChars: [] }`.
+   * @param options Either the number of characters or an options instance. Defaults to `{ count: 1, casing: 'lower', bannedChars: [] }`.
    * @param options.count The number of characters to generate. Defaults to `1`.
-   * @param options.upcase If true, the result will be uppercase. If false, it will be lowercase. Defaults to `false`.
+   * @param options.casing The casing of the characters. Defaults to `'lower'`.
```
[the remark this answers is not in the document]
Comment (Shinigami92) [reply src/modules/random/index.ts:155]: come from ts :+1:

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**